tests/abi/tools/revng-check-compatibility-with-abi/ABIArtifactParser.cpp

```diff
@@ -192,7 +193,11 @@ State::Parsed abi::artifact::parse(llvm::StringRef RuntimeArtifact,
   revng_assert(!Reader.error());
 
   llvm::StringRef ArchitectureName = model::Architecture::getName(Architecture);
-  revng_assert(ArchitectureName == Deserialized.TargetArchitecture);
+  std::string E = "Target architecture ('"
+                  + Deserialized.TargetArchitecture.str()
+                  + "') does not match the expected one: '"
+                  + ArchitectureName.str() + "'\n";
+  revng_assert(ArchitectureName == Deserialized.TargetArchitecture, E.c_str());
```

Comment:
* You seem not to like `Twine`, any reason for that?
* Consider that if you leave this out of the assert, the string will always be computed, even if the assert is not triggered and (likely) even in release.

Reply:
1. Literally every experience I even had with it was godawful. And considering it's going to get stringified anyway (since the assert expects a zero-terminated c-string as far as I understand), I thought that I might as well do so manually.

2. Yeah, that is pretty sad, but I couldn't manage to get it any readable any other way.

But now that I think about it, I feel like `llvm::formatv` (and eventually `std::format` once we update `libc++`) could be solution we both feel fine about, consider:
```cpp
revng_assert(ArchitectureName == Deserialized.TargetArchitecture,
             llvm::format_v("Target architecture ('{0}') does not match "
                            "the expected one: '{1}'\n",
                            Deserialized.TargetArchitecture.str(),
                            ArchitectureName.str())
                .str().c_str());
```

What's your take?

Reply:
I'm not a super fan of `formatv`, I don't like string interpolation with variables at the end instead of interleaved.
Personal opinions apart, I prefer to preserve the coherence of the codebase in view of a day of refactoring, so let's stick with `Twine` or string concatenation.

If you can't fit in a single line, just use `if (Condition) { /* Prepare string */; revng_abort(String); }`.

Reply:
`if (Condition)` sounds good!

Reply:
Done:
https://github.com/revng/revng/pull/232/commits/d80da836b3f6ccf5804e33b2667ee70c9dcb11b3
https://github.com/revng/revng/pull/232/commits/c9db2ea14eb3c45df9d058b275c836a7e9236443
https://github.com/revng/revng/pull/232/commits/69ee629b81560a8cfad7bd779873e7d0c7b45090